### PR TITLE
[awslogs] Auto-detect region and set user-agent

### PR DESCRIPTION
**Pull request by samuelkarp**
This pull request enhances the `awslogs` logging driver such that it can automatically detect the correct Amazon CloudWatch Logs region when the Docker daemon is running on an Amazon EC2 instance.

### CODE REVIEW

**Comment by samuelkarp** [comment]
Any update here?

**Comment by thaJeztah** [comment]
Hi @samuelkarp apologies for the delay, it's been busy times, so reviews are taking a bit longer than we want.

ping @calavera I saw you reviewed the original PR, perhaps you can have a look at this?

File: daemon/logger/awslogs/cloudwatchlogs.go [hunk 1]
```diff
@@ -100,6 +110,52 @@ func New(ctx logger.Context) (logger.Logger, error) {
...
+		})
+	}
+	if config.Region == nil || *config.Region == "" {
+		logrus.Info("Trying to get region from EC2 Metadata")
+		ec2MetadataClient := newRegionFinder()
+		region, err := ec2MetadataClient.Region()
+		if err != nil {
+			logrus.WithFields(logrus.Fields{
+				"error": err,
+			}).Error("Could not get region from EC2 metadata, environment, or log option", err)
```
Comment (calavera) [inline daemon/logger/awslogs/cloudwatchlogs.go]: did you mean to use `Errorf` here? It's going to print an unusual message if we leave it like this.

**Comment by calavera** [comment]
I minor comment about error handling, otherwise LGTM.

**Comment by tiborvass** [comment]
LGTM

File: docs/reference/logging/awslogs.md [hunk 2]
```diff
@@ -34,9 +34,10 @@ You can use the `--log-opt NAME=VALUE` flag to specify Amazon CloudWatch Logs lo
 
 ### awslogs-region
 
-You must specify a region for the `awslogs` logging driver. You can specify the
-region with either the `awslogs-region` log option or `AWS_REGION` environment
-variable:
+The `awslogs` logging driver must send your Docker logs to a specified region;
```
Comment (moxiegirl) [inline docs/reference/logging/awslogs.md]: The phrasing is a bit odd.  Maybe;

The `awslogs` logging driver sends your Docker logs to a specific region. Use the `awslogs-region` log option or the `AWS_REGION` environment variable to
set the region value. By default, if your Docker daemon is running on an EC2 instance and no region is set, the driver uses the instance's region.

**Comment by samuelkarp** [comment]
@calavera and @moxiegirl, thanks for the review!  I've rebased on master and addressed both your comments.  (I'm seeing some failures in Jenkins, but I believe those to be unrelated to my code.)

**Comment by thaJeztah** [comment]
@samuelkarp thanks! docs LGTM

ping @moxiegirl

**Comment by calavera** [comment]
Failures are unrelated, but I'll wait for @moxiegirl to give her :+1: before merging.

**Comment by moxiegirl** [comment]
LGTM! Thank you @samuelkarp

**Comment by samuelkarp** [comment]
Thanks!

**Pull request merged**